src/vchan_cmds.c

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2024 EPAM Systems
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <domain.h>
+#include <vch.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/shell/shell.h>
+
+LOG_MODULE_REGISTER(vchan_cmd);
+
+#define ECHO_SERVER_THREAD_PRIO    7
+static struct k_thread echo_thread;
+static struct k_thread *pecho_thread = NULL;
+struct vch_handle handle;
+
+static K_THREAD_STACK_DEFINE(main_stack, 8192);
+static volatile int vchan_server_stop = 0;
+static char buf[10];
```

Comment:
should this variables be global? as I can see you are using them only in single function
also their size choices are not clear

Reply:
Size is configurable. buffer declaration moved to function.